src/openqaoa-core/backends/cost_function.py

```diff
@@ -0,0 +1,78 @@
+# Cost function to be used for QAOA training
```

Comment:
I think we could call this file 'helper_functions.py' or something more general than 'cost_function.py'

Reply:
Since the methods here are purely related to how the cost values are computed. I believe `cost_function.py` is a more accurate name for now.